src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,445 @@
+        {
+            // Arrange
+            var dataSourceItem = new DataSourceItem();
+            var visualization = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true }.Object;
+            var testDescription = "testDescription";
+
+            // Act
+            visualization.Description = testDescription;
+
+            // Assert
+            Assert.Equal(testDescription, visualization.Description);
+        }
+
+        [Fact]
+        public void FilterBindings_EqualDataDefinitionBindings_WhenUsed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                    new TextField("C"),
+                    new TextField("D"),
+                }
+            };
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            // Act
+            var filterBindings = visualization.FilterBindings;
+
+            // Assert
+            Assert.Equal(((DataDefinitionBase)visualization.DataDefinition).Bindings.Bindings, filterBindings);
+        }
+
+        [Fact]
+        public void GetDataSourceItem_ReturnsCorrectDataSourceItem_WhenUseTabularData()
+        {
+            // Arrange
+            var dataSource = new DataSource()
+            {
+                Id = "DataSourceId",
+            };
+            var dataSourceResourceItem = new DataSource()
+            {
+                Id = "dataSourceResourceItemId",
+            };
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                    new TextField("C"),
+                    new TextField("D"),
+                }
+            };
+            dataSourceItem.ResourceItem = new DataSourceItem()
+            {
+                DataSource = dataSourceResourceItem,
+                Id = "ResourceItem",
+            };
+
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            var document = new RdashDocument();
+            document.DataSources.Add(dataSourceResourceItem);
+            document.Visualizations.Add(visualization);
+
+            RdashDocumentValidator.Validate(document);
+
+            // Act
+            var dataSourceItemResult = visualization.GetDataSourceItem();
+
+            // Assert
+            Assert.Equal(dataSourceItem, dataSourceItemResult);
```

Comment:
Need Assertions to covers the code below:

> if (DataDefinition is TabularDataDefinition tdd)
> {
>     tdd.Fields = dataSourceItem.Fields.Clone();
>     
>     if(tdd.DataSourceItem.JoinTables != null && tdd.DataSourceItem.JoinTables.Count > 0)
>     {
>         tdd.JoinTables = tdd.DataSourceItem.JoinTables.Clone();
>     }
> }

Reply:
JoinTables is asserted in another test already